src/libs/Altinn.Authorization.Host/src/Altinn.Authorization.Host.Lease/StorageAccount/StorageAccountLease.cs

```diff
@@ -1,177 +1,224 @@
+                return await BlobClient.UploadAsync(stream, options, cancellationToken);
+            });
+        }
+        catch (RequestFailedException ex) when (ex.ErrorCode == BlobErrorCode.LeaseLost || ex.ErrorCode == BlobErrorCode.LeaseAlreadyPresent)
+        {
+            Cancel();
+        }
+        finally
+        {
+            Semaphore.Release();
         }
-
-        return lease;
     }
 
     /// <inheritdoc/>
-    public async Task Release<T>(LeaseResult<T> lease, CancellationToken cancellationToken = default)
-          where T : class
+    public async Task Update<T>(Action<T> configureData, CancellationToken cancellationToken = default)
+        where T : class, new()
     {
-        if (lease.HasLease && lease is StorageAccountLeaseResult<T> castedLease)
+        if (configureData is { })
         {
-            try
-            {
-                await castedLease.BlobClient.GetBlobLeaseClient(castedLease.Response.LeaseId).ReleaseAsync(default, cancellationToken);
-                castedLease.Response = null;
-            }
-            catch (RequestFailedException ex) when (ex.ErrorCode == "LeaseLost" || ex.ErrorCode == "LeaseAlreadyPresent")
-            {
-                Log.FailedToReleaseLease(Logger, ex);
-            }
+            var data = new T();
+            configureData(data);
+            await Update(data, cancellationToken);
         }
     }
 
     /// <inheritdoc/>
-    public async Task<LeaseResult<T>> RefreshLease<T>(LeaseResult<T> lease, CancellationToken cancellationToken = default)
-        where T : class
+    public async ValueTask DisposeAsync()
     {
-        if (lease.HasLease && lease is StorageAccountLeaseResult<T> castedLease)
+        if (Interlocked.Exchange(ref _disposed, 1) == 1)
+        {
+            return;
+        }
+
+        try
         {
-            var leaseExpirationTime = castedLease.Acquired.AddSeconds(Convert.ToDouble(castedLease.Response.LeaseTime));
-            if (leaseExpirationTime < DateTime.UtcNow.AddSeconds(20))
+            await LeaseRefresherCancellation.CancelAsync();
+            if (RenewalTask is { })
             {
                 try
                 {
-                    var result = await castedLease.LeaseClient.RenewAsync(default, cancellationToken);
-                    castedLease.Response = result;
-                    return castedLease;
+                    await RenewalTask.WaitAsync(TimeSpan.FromSeconds(5));
                 }
-                catch (RequestFailedException ex) when (ex.ErrorCode == "LeaseLost" || ex.ErrorCode == "LeaseAlreadyPresent")
+                catch
                 {
-                    Log.CantRefreshLease(Logger, ex);
-                    castedLease.Response = null;
-                    return castedLease;
                 }
             }
+
+            var release = ReleaseLease(CancellationToken.None);
+            await release.WaitAsync(TimeSpan.FromSeconds(5));
+        }
+        catch
```

Comment:
tomme catch og final statements

Reply:
Kan være greit med en kommentar i tom catch dersom den beholdes slik med vilje for å håndtere gitt case.